.travis.yml

```diff
@@ -1,32 +1,32 @@
-language: cpp
+language: go
+
+sudo: required
+dist: trusty
+
+go:
+  - "1.10.x"
 
 addons:
   apt:
     packages:
       - flawfinder
-      - python-pip
       - squashfs-tools
       - uuid-dev
+      - libuuid1
       - libssl-dev
+      - libssl1.0.0
       - libarchive-dev
-
-env:
-  global:
-    - secure: ZgH9zeVLyn7BvZo5IiVlBsmteLi1qsTwlCA6/60iIG+WJ1S1cbYRQeHyVvrat+wdjXqcJxRGyg/J6iGAFObIqngzSHPutyV48KgGV8tdPM0Rr+vtEJgW4b8pVcrlI2lsjgG1avqkW93SyOiaqWNE9cIQhlOKNEvIP2exqEuTra0muxl8rOWMpLTdKpA8NPsB0Xx+gRVZvwLtmSjqaSjNDrv/jt+HsnKazZQYKoewB+WO7hT/A4u0TcE+cggEep8LzjZt1xiU9O+OSHlibiKeYLDex+Vl6VtkNrlWYemFQR37nUJaleSXPFQ7H7DETmLCQtQ3NZSlrW/c+mwqFxJTfvgJ6xgK5fp7FlwMpqLBlgZrYAgyxSSC/N7o49zugYlJyoVKk9rDWGLRrxrmoX99hizsEE9bp6hyn9sXbBjsbzmQwhUOChiVAuEbGuLAYNyZIXbhfjbDUJVz748MQS0ymqK3jLrZOYTEs1/8IVk02FhRVKBdy6jHPbgD7egwls3P41jUgpmGxOLl9lvY/7lUo4ckrSso6jQELDIUcpyQiZxlOAhoic1sY6/vKqHB/RxUViufE/OdCDumwNcDfP8X8mpIeDUNn0ECogpx4ChpzdB/1c0YRRNT0O3E7qaVmyAApHCKcnZyskFWHr2FvCU2ToejgZxCqeP0YIdPMeROntA=
-    - secure: "mOVQKTzAfpcemi9mLLts0h1aU2FhhNwurwkwiLBjiLYql+QgBX2wMWsCiVYx10trYil3jFCyEuPB4Zu84ukInSpb/QvKcBT7N1Cqpdkw2qrc2faPThPrmwyJ5gg7wb70sg0Fiz3eKY2fe1f6ab7ZNA+prvttskz4YKskkGIJoG6jH1JpD+T8jixozGSp3rgu4FkvMwkq4zzZ53R/8+eANfcleTA0evfPXfXv+n1nQjTMTCrsHeThzkXad/PZjqR4vMcCKdQ4g31B8uX17gOtoQvhe6jHCABFW5qzJ/Fgj6CE1LiEt6sLk0pwMjJ7vsbgdhgwveQker14CN5PumBNr08VFutpfJ/eEqVw2mIONUx6sfPExNa4V+xPTdXXVaLs55UiXOpi84FlNqn6AlmMMBQryDPMRcxudspXboHLReMP5fnikOjs1zBNJlUKoMFkcmR5A5jgZYwHE1WONVNevHM3wuhPaAjds4jxFPqckH6nTe3Zl5CTKRV4N+ieTd30y6dFcl1PXLFeE9kr2hH0PdCuxHK+W9zM/9nj8PZWhVyNe40OZl/Cvm4KNvTikSWGjzLlfaAlsgNXoxRxLpE7EniH+ZUgiEgbbxPETT4+z1W7uzIjJHPxjfJeImNAy2eMZ3CVeo4ZUuxt+f8rLGsyAkVhpj9ZQDPnICNoBV8H304="
-
-sudo: true
-
-matrix:
-  include:
-    - os: linux
-    - os: linux
-      env: OS_TYPE=centos OS_VERSION=6
-    - os: linux
-      env: OS_TYPE=centos OS_VERSION=7
+      - libgpgme11-dev
 
 before_install:
-  - .travis/before_install
+  - curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+  - export PATH="${GOPATH}/bin:${PATH}"
+  - dep ensure -vendor-only
+
+install:
+  - ./mconfig -b ./buildtree -p /usr/local
+  - make -j `nproc 2>/dev/null || echo 1` -C ./buildtree
+  - sudo make -C ./buildtree install
 
 script:
-  - .travis/script
+  - go test ./tests/
```

Comment:
I'd strongly encourage adding `go fmt`, `go vet`, and `golint` in the CI (and perhaps during the `make`.)

`go fmt ./...` is clean in your branch and could easily be added as part of this PR. See https://github.com/sylabs/remote-build/blob/master/.circleci/config.yml for an example (it annoyingly doesn't set a non-successful return code if formatting problems are found.)

`go vet` and `golint` are both finding issues and might be best addressed separately (I have added comments for new `go vet` issues introduced by this PR.)